common/src/main/java/com/tc/net/protocol/transport/ConnectionHealthCheckerImpl.java

```diff
@@ -44,18 +46,24 @@
 public class ConnectionHealthCheckerImpl implements ConnectionHealthChecker {
 
   private final Logger logger;
-  private final Timer                           monitorThread;
+  private static final ScheduledExecutorService monitorThread = Executors.newSingleThreadScheduledExecutor((r) -> {
+    ThreadGroup grp = Thread.currentThread().getThreadGroup();
+    while (grp instanceof TCThreadGroup) {
+      grp = grp.getParent();
```

Comment:
why it cannot be in a TcThreadGroup ?

Reply:
because the client shutdown code will hang on threads that it believes belong to it.  This thread no longer belongs to a specific connection but all the connections